src/userDashboard/userdashboard_Graphs/dataFormat/TranslatedWordsDataFormat.js

```diff
@@ -0,0 +1,285 @@
+import {
+  subDays,
+  format,
+  getYear,
+  getMonth,
+  eachDayOfInterval,
+  subMonths,
+  subYears,
+} from "date-fns";
+import {
+  PERIOD_OPTIONS,
+  DATE_FORMAT,
+  LINE_GRAPH_BOTTOM_LEGEND,
+} from "../../ConstantsUserDashboard";
+import { calculatePerMonth } from "./CommonDataFormat";
+
+/**
+ *
+ * Function that transforms an array of pairs to
+ * a map of type key->value
+ *
+ * To be used for TranslatedWordsGraph
+ *
+ * @param {Array} data
+ *
+ * Array containing pairs of dates and second counts
+ * as received from the API for the number of translated words for user
+ *
+ * [{date:dateString1, count: secondsCount1}, {date:dateString2, count: secondsCount2}, {date:dateString3, count: secondsCount3}, …]
+ *
+ * example:
+ * (451)[{…}, {…}, {…}, {…}, {…}, {…}, {…}, …]
+ *
+ * example entries:
+ * 0:{date: '2018-03-02', count: 1223}
+ * 1:{date: '2018-03-03', count: 2534}
+ * 2:{date: '2018-03-05', count: 5498}
+ *
+ * @returns {Map}
+ *
+ * Returns a map with the date values as key and seconds count as value
+ * Map() {date1 => count1, date2 => count2, date3 => count3, …}
+ *
+ * example:
+ * Map(451) {2018-03-02 => 1223, 2018-03-03 => 2534, 2018-03-05 => 5498, 2018-03-06 => 5991, 2018-03-07 => 2935, …}
+ */
+function getMapData(data) {
+  var result = new Map();
+
+  data.forEach((row) => {
+    result.set(row.date, row.count);
+  });
+
+  return result;
+}
+
+function getDataForInterval(data, startDate, endDate, dateFormatString) {
+  var result = [];
+
+  var dates = eachDayOfInterval({ start: startDate, end: endDate });
+
+  dates.forEach((day) => {
+    var stringDate = format(day, DATE_FORMAT);
+    var stringlegend = format(day, dateFormatString);
+    data.has(stringDate)
+      ? result.push({ x: stringlegend, y: data.get(stringDate) })
+      : result.push({ x: stringlegend, y: 0 });
+  });
+
+  return result;
+}
+
+function getLineDataForWeek(data, dateInWeek) {

```

Comment:
there is a lot of duplication between `getLineDataForWeek` and `getLineDataForMonth`. taking the number of days as parameter, and renaming `dateInMonth` and `dateInWeek` to 'endDate' would probably allow you to reuse this function. or?

Reply:
done.